### ISSUE TITLE
Redesign - EFY

### ISSUE THREAD

**Issue by dragos-efy** [TeamPiped/Piped#1391]
Can be optimised more obviously, but here's the new Piped redesign! 🥳

For people that are not part of the matrix room, it uses [efy ui](https://github.com/dragos-efy/efy) to add more customisation features, so things changed a bit, but I'm working on the documentation to make it easier for devs to understand the logic behind it and contribute. Any feedback or suggestions you have, please share them, I and other really cool people who I'm grateful for have been testing it, but bugs can still come up, therefore if you wanna share something, please do!

Live demo: https://efy.piped.pages.dev

Special thanks to: Bnyro, Kavin, Zucc (whatever.social), Diamond_AaronXG and other small or anonymous contributions, you guys rock! 🔥

**Comment by Bnyro** [comment]
The code for it can be found at the `efy` branch if anyone is willing to test it locally or do anything else with the code :)

**Comment by Senor-Ducky** [comment]
Bug: The search bar appears to be shrunk and not working right.
![image](https://user-images.githubusercontent.com/61725851/188673018-f755b3f9-7184-4019-a9ac-95dd937f8a1d.png)

**Comment by dragos-efy** [comment]
@Senor-Ducky It got broken after the last commit, it was okay before, but I think I identified the cause for the most part. Thank you! If you notice other bugs lmk

**Comment by dragos-efy** [comment]
The live instance will likely have issues for a while, but the css code is fine, it's just needs to be split and distributed to vue components. Keep sending feedback, but keep in mind that there might be a difference between how unstable it seems to be and how stable it actually is

**Comment by IronMaltese** [comment]
![image](https://user-images.githubusercontent.com/89056889/188745034-e5e8fafb-1a09-488c-982a-879b71a5f696.jpeg)

The arrow seems to be too close to the word

**Comment by dragos-efy** [comment]
@IronMaltese Glad you like it! Your issue was on the list too, but having it here makes it easier to keep track of what I gotta do lol, so thanks!

**Comment by dragos-efy** [comment]
Everyone and @Senor-Ducky , if you tried the live instance before the problem got fixed, try clearing your cache (`CTRL+F5` on some browsers) or opening it in a private tab.  And remember, if you need to clear, import or export your local storage or background images, go to `Save & Restore` in the efy menu. Things should be as stable as they were before + a few small issues that I'll be working on when I have time. Planning to fix all the issues mentioned here in the next 2-3 days or sooner. Enjoy & keep giving feedback! 🥳

**Comment by dragos-efy** [comment]
Fixed all the issues above, refresh your cache if needed!

**Comment by martin-desktops** [comment]
Found a bug: while video details are loading the video player doesn’t have rounded corners for a short time.
![BD54A6CD-2CD8-49AC-A9B1-18F2C30E5247](https://user-images.githubusercontent.com/64419373/190874290-a278c322-0c93-4d21-bbeb-815161d4ccbb.png)


What a great job btw ^^

**Comment by martin-desktops** [comment]
Bug number 2.: the font used in the new design doesn’t support some polish letters what makes them looking out of place
![33C110E9-EC89-4A7E-BA8D-7ED1B7C4135C](https://user-images.githubusercontent.com/64419373/190874363-0b5beeb3-e31f-4616-85f2-e80dc98d4f1d.png)

**Comment by martin-desktops** [comment]
And I hope the last one: active search input field doesn’t have rounded focus outline
![509D02AB-AE5F-4647-93B2-828F2352AEDD](https://user-images.githubusercontent.com/64419373/190874478-67cd8c5e-ac0c-422b-b41e-7ed7c1c2d44d.jpeg)

Tested on Safari, iOS 15.5

**Comment by EdwardLangdon** [comment]
![Uploading image.png…]()

The Text is too big in this modal. Would be better if it is smaller.

**Comment by EdwardLangdon** [comment]
![image](https://user-images.githubusercontent.com/101938856/191040048-570cd920-26e0-491f-bdc4-5f79bafa63ef.png)

Better to align these elements.

**Comment by EdwardLangdon** [comment]
- [] Apply red colour, if video is tagged as `Shorts` and Black colour for `Watched`. As giving same colour to both often confuses.

**Comment by opl-** [comment]
The animation which plays on click is extremely distracting. There should be a way to disable it.

**Comment by dragos-efy** [comment]
@martin-desktops 
> active search input field doesn’t have rounded focus outline

fixed in the new update. I'll look into the rounded corners for the video and polish characters. Thanks for the feedback!!

**Comment by dragos-efy** [comment]
@EdwardLangdon thanks again! I'll look into all of them. For the shorts tho, I think it could be the main colours rather than red, so that it fits with whatever colours the user chooses and instead of black, the variable for the background, like the time on the video, which can be black, white, chosen by the user etc. So basically kinda the same as your idea, but with custom colors

**Comment by dragos-efy** [comment]
> The animation which plays on click is extremely distracting. There should be a way to disable it.

Alright @opl- , I can add a new accessibility option for that then. I need to figure out how to implement it, but it should be fairly easy I think. Thank you!

To make sure I fully understand you, do you mean buttons or any animation related to clicks in general?

**Comment by opl-** [comment]
The main thing for me was the scale down -> scale back up "pulse" animation playing whenever I clicked on any button, video or any other interactive element.

**Comment by dragos-efy** [comment]
@opl- Alright, got you then

**Comment by dragos-efy** [comment]
@opl- There will be a toggle inside `efy sidebar` > `accessibility` > `animations` in the next update. I made it work.

**Comment by martin-desktops** [comment]
Hello! 

I am back with one new reflection: subtitles aren't legible in some cases right now. Sometimes there is no enough contrast between the light text and blurred video, as it is shown on the screenshot below. 

My recommendation is to darken the blurred background.
![image](https://user-images.githubusercontent.com/64419373/198391602-5c9bff05-3221-4b22-b967-91713d812898.png)

**Comment by dragos-efy** [comment]
@martin-desktops Thanks again! We're planning to make a module in the sidebar for captions, so that you can choose between multiple styles, like **maybe** (not decided yet), you can choose between background, no background and blurred. Right now I'm trying to solve issues that get the framework and redesign to a stable state and then things like captions or comments can become more modular and doing that should take less time. I'm pretty limited time wise by having to apply to jobs, which is why things have been slower, but I acknowledge the current problem with captions or other similar bugs, like the white flash on page load. Keep the feedback going when you need to! Will take care of this as soon as I can!

Btw, I can't replicate your bug with Polish characters, so please let me know how I can if it's still present.

**Comment by pc00per** [comment]
Just out of curiosity. Is https://piped.mha.fi the same redesign as this development thread ?

**Comment by Akczht** [comment]
People! is it just me or do you also hate the way the player situation is in ios , seriously who wants this 
![Uploading 67A4E2AE-2FC6-4EDB-AE6C-809887194129.png…]()

**Comment by Akczht** [comment]
![Uploading 35C1CF7A-9889-492F-9B05-9E3A1E03C632.png…](It’s Utly Right ?)

**Comment by Akczht** [comment]
People! is it just me or do you also hate the way the player situation is in ios , seriously who wants this 
![2838F4CC-F08A-4744-B1BC-C69F0A018A2B](https://user-images.githubusercontent.com/79302042/199080856-59c3748d-f8b1-4252-a257-3a469f75569f.png)

**Comment by Bnyro** [comment]
> People! is it just me or do you also hate the way the player situation is in ios , seriously who wants this 
> 
> EDIT: I was to ask are there any plans for changing it ? or is it just way it was intended to be , because it is hard for me to navigate the player UI and I don’t see the point of having two players , rather the native iOS Safari player can be used for things like subtitles and playback speed , as for the quality change button it can have a dedicated button , this is the best I can say as I’m not a dev
> 
> Thanks!
> 
> ![2838F4CC-F08A-4744-B1BC-C69F0A018A2B](https://user-images.githubusercontent.com/79302042/199080856-59c3748d-f8b1-4252-a257-3a469f75569f.png)
> 

If I may ask, how is that related to EFY? There are already open issues because of some issues concerning iOS support you could have commented on, but that one here is just meant to discuss potential bugs or requests concerning Piped ft. EFY, so your comment feels kinda out of place here. Even the screenshot you posted doesn't use the EFY version of Piped, so it's definitely not an issue related to it!
Please just move over to a corresponding issue or open a new one if you can't find one, thanks!

**Comment by Bnyro** [comment]
> Just out of curiosity. Is https://piped.mha.fi the same redesign as this development thread ?

Yes, mha.fi already uses uses the redesigned version built with EFY.

However, doing so is not recommended yet as its still in development and might contain bugs the current interface doesn't, so the instance owner acts on their own risk ;)

**Comment by Akczht** [comment]
> > People! is it just me or do you also hate the way the player situation is in ios , seriously who wants this
> > EDIT: I was to ask are there any plans for changing it ? or is it just way it was intended to be , because it is hard for me to navigate the player UI and I don’t see the point of having two players , rather the native iOS Safari player can be used for things like subtitles and playback speed , as for the quality change button it can have a dedicated button , this is the best I can say as I’m not a dev
> > Thanks!
> > ![2838F4CC-F08A-4744-B1BC-C69F0A018A2B](https://user-images.githubusercontent.com/79302042/199080856-59c3748d-f8b1-4252-a257-3a469f75569f.png)
> 
> If I may ask, how is that related to EFY? There are already open issues because of some issues concerning iOS support you could have commented on, but that one here is just meant to discuss potential bugs or requests concerning Piped ft. EFY, so your comment feels kinda out of place here. Even the screenshot you posted doesn't use the EFY version of Piped, so it's definitely not an issue related to it! Please just move over to a corresponding issue or open a new one if you can't find one, thanks!

got it

**Comment by PrivacyDevel** [comment]
Hey first of all, thank you for the nice new Piped redesign! :)

One little bug that I just noticed is that it currently always shows "Comments are disabled in the settings" right above the actual comments and under the video description.

**Comment by Bnyro** [comment]
> Hey first of all, thank you for the nice new Piped redesign! :)
>
I agree, @dragos-efy did a great job on that :)

> One little bug that I just noticed is that it currently always shows "Comments are disabled in the settings" right above the actual comments and under the video description.
>
Thanks for mentioning, should be fixed with  #1685 :)

**Comment by DeadKper** [comment]
![Screenshot_2022 11 05_14 20 11](https://user-images.githubusercontent.com/35360914/200141796-5f1867ce-e9a8-4a19-a5ba-681be6f5f09d.png)

Shouldn't border radius be 0 on full screen or is it like that on purpose? Didn't notice until now since most videos don't have a really bright background and I don't tend to look at the corners that often

**Comment by git-Hmmm** [comment]
I didn't like the accent color implementation also it shows very off animations on clicking or something, I hate the overall concept tho some buttons are designed properly. According to me just the normal pipe needs some css improvements and that's it, this efy ui doesn't satisfies me.

**Comment by FireMasterK** [comment]
I'll note down a few issues here:

- All audio files are loaded at once (not on demand)
- Audio files are loaded even if audio isn't enabled
- Having cookies disabled/localstorage, results in a broken theme
- The website briefly appears white then turns dark when loading, ideally, we need to have some basic CSS to default to a dark theme
- The service worker does not cache fonts and Mp3 files (since they're not part of the source code and are not preloaded, maybe we need runtime caching?)
- Webm/Opus could be used instead of mp3?

**Comment by dragos-efy** [comment]
Fixed all of these, reload the cache and try them out (on efy.piped.pages.dev , not sure about other instances):

@IkelAtomig
> - Better to align these elements
> - shorts color

@DeadKper
> Shouldn't border radius be 0 on full screen?

@martin-desktops : Captions can now have no background or a black one as well. It's in alpha testing, so go to `efy sidebar` > `Piped Style` > `Captions`. The changes won't save, and it will be more organised than now, but you can try it out in the meanwhile. The darkness of the blur is increased by 20% now btw. Lmk how it goes!

**Comment by dragos-efy** [comment]
> I'll note down a few issues here:
> 
> * All audio files are loaded at once (not on demand)
> * Audio files are loaded even if audio isn't enabled
> * Having cookies disabled/localstorage, results in a broken theme
> * The website briefly appears white then turns dark when loading, ideally, we need to have some basic CSS to default to a dark theme
> * The service worker does not cache fonts and Mp3 files (since they're not part of the source code and are not preloaded, maybe we need runtime caching?)
> * Webm/Opus could be used instead of mp3?

Thanks!! For the sounds, it looks like sometimes they load, sometimes not... so idk, they're supposed to be loaded when needed... I'm open to suggestions if you find solutions.

I'll work on the localStorage, white flash and webm/opus soon, although I'll chill for about 1 week I think to have more time to apply to jobs, since I'm really tight on money right now. But yeah, was about to convert the mp3 files anyway to save space, good take!

And for runtime caching idk, maybe? You're better at that sort of stuff so I guess we can try multiple things and see what works better. I'd just prefer to not need to split files a lot or require a server version of the same files for efy (I already did that with a few files, but I want the number to not increase lol), unless it can be done with minimal effort so that it's easier to maintain, but otherwise I'm open to all sorts of suggestions.

I agree with all the stuff you mentioned. We'll do it!

**Comment by IkelAtomig** [comment]
Checked as expected.

**Comment by DeadKper** [comment]
![Screenshot_20221108-212450_Brave](https://user-images.githubusercontent.com/35360914/200859831-527fdd8c-db18-420d-b60e-2fb79938a040.jpg)
![Screenshot_20221108-212409_Brave](https://user-images.githubusercontent.com/35360914/200859838-90fd412e-8bd0-4936-b086-e21f7a8f4c31.jpg)

There's 1 bug on cellphone, you can see it on the screenshot, maybe only changing border radius of the video to 0 could be a solution?

Also 2 improvements that can be made for phones

- Double tap on the left/right side to jump 10 secs back/ahead
- Center the background when the width is smaller than the height

Also, changing any settings on "Visual Filters > Trans Elements" makes the page really slow on phones, the desktop doesn't seem to have any issues so far

**Comment by Bnyro** [comment]
Concerning the double tap, that's nothing that would be part of the redesign but rather an update of the shaka-player which we are using, see https://github.com/shaka-project/shaka-player/pull/4649.

**Comment by dragos-efy** [comment]
@DeadKper
> maybe only changing border radius of the video to 0 could be a solution?

Yup, that's the solution, I'm fixing this next!

> changing any settings on "Visual Filters > Trans Elements" makes the page really slow on phones, the desktop doesn't seem to have any issues so far

Yeah, usually cuz blur and `backdrop-filter` (the css property used for the trans elements) is slower on phones than `filter` (the css property used for content and background, much faster), probably cuz they usually don't have fast GPUs like on PCs / laptops / handhelds. For some users it will be snappy and for others slow. Use what works for you, the options are there in case you want / need them to have the freedom to do it, but if the hardware / browser doesn't support it well, that's alright. You're already customising things more than other apps would allow you to anyway, as I see you have a custom background and border radius. Cool! 🥳

I'll say tho, changing browsers can help sometimes... For example Gnome web is the slowest I've tried. Firefox mobile is a bit snappier than brave mobile in some aspects and on desktop, brave has less visual glitches, but firefox uses hardware acceleration better sometimes so it results in more speed, but not when it comes to heavy animations. I guess try multiple variants and see if you can increase the performance or not. If you have any discoveries, let me know, I'd be curious to hear them.

> Center the background when the width is smaller than the height

Uhmm, I guess it makes sense, it's probably not gonna be a priority right now tho, but I'll look into it for sure when my time allows it! Thanks for the feedback, really helpful!

**Comment by dragos-efy** [comment]
@Bnyro > On mobile, when clicking the hamburger menu and go to playlists after while not being logged in the account dialog pops up but can't be dismissed, neither by clicking the x nor in the void

Yeah, cuz that's what the playlists page does haha. I just redirect people there even if they're not logged in lol. Which is why I said it must be done dynamically. Or make the playlists page work for both user states with some `v-if` conditions maybe, then I guess it won't need to be dynamic from the menu... whatever you feel is better

**Comment by Bnyro** [comment]
> Yeah, cuz that's what the playlists page does haha. I just redirect people there even if they're not logged in lol. Which is why I said it must be done dynamically. Or make the playlists page work for both user states with some `v-if` conditions maybe, then I guess it won't need to be dynamic from the menu... whatever you feel is better
>
Oh, true, that's the same thing as for the current frontend and not related to EFY, my bad.
But I think now we can just show the login dialog instead of redirecting to that page in order to make it more user friendly, hence users will be able to still navigate on Piped without being logged in and don't get stuck in an unresponsive dialog.
I will make a PR for that later :)

**Comment by dragos-efy** [comment]
cool

**Comment by DeadKper** [comment]
![image](https://user-images.githubusercontent.com/35360914/201273801-8cc84a34-c749-4bf8-b16e-f8b8ccb0dd8b.png)
Visual filters are not applying correctly on the video chapters, you can see in the screenshot the difference between the comments and the chapter, also, great work with the design I'm loving it

**Comment by dragos-efy** [comment]
@DeadKper glad you love it, it's fixed now 🤟 as well as the fullscreen player, but you probably noticed that

**Comment by photochlorination** [comment]
A minor issue, depending on the window width and view count the add to playlist button might wrap onto a new line, which is somewhat distracting.
![image](https://user-images.githubusercontent.com/86535034/201433813-c3215268-b521-4179-9d60-7acbddcb33a9.png)
(Firefox at 2560 window width)

**Comment by DeadKper** [comment]
![image](https://user-images.githubusercontent.com/35360914/201504020-084d7b77-9d34-4363-b2a5-963e00cbb44a.png)
![image](https://user-images.githubusercontent.com/35360914/201504028-6aba079b-2345-4cc4-a4c0-4d91cd23c01a.png)
Video doesn't load, reloading page with f5 or ctrl + f5 only gives a different error, deleting service workers and realoding fixes it

**Comment by dragos-efy** [comment]
@photochlorination I could add a minimum height which makes them all the same, but then all cards would be tall even when not needed. I'm keeping this on hold for now unless you have practical ideas to suggest, although if there's enough interest, I can maybe add an alpha testing option in the sidebar for users to try out multiple variations. Thanks!

**Comment by dragos-efy** [comment]
@DeadKper I know it sounds weird after you mentioned all the intense stuff you tried haha, but have you tried doing it in a private tab? The error is gone for me when I do that or `ctrl + f5` on a video page. I spotted the same bug, but when entering a video link (like /watch=idofvideo), otherwise directly accessing videos by clicking on them seems fine. Also some of the errors in the console are for new extra modules, not the main framework, so you can safely ignore them for now until we make them work. Looks like it's related to the piped `WatchVideo` vue component probably, which I modified quite a bit last time, so maybe something's wrong there, OR just some caching issue, since for me private tabs and `ctrl + f5` work... not sure yet. The extensions are not the problem anyway.

@Bnyro @FireMasterK any ideas?

**Comment by martin-desktops** [comment]
Hello, I don't quite understand the reasoning behind making video information (creator, views, duration) looking exactly like buttons. It's a bit misleading for me.

![Screenshot from 2022-11-13 14-20-33](https://user-images.githubusercontent.com/64419373/201523822-d088a22e-0212-4745-9046-e2e603901468.png)

**Comment by photochlorination** [comment]
> @photochlorination I could add a minimum height which makes them all the same, but then all cards would be tall even when not needed. I'm keeping this on hold for now unless you have practical ideas to suggest, although if there's enough interest, I can maybe add an alpha testing option in the sidebar for users to try out multiple variations. Thanks!

I don't think it is necessarily the height that is the problem; what I found distracting was the wrapping of the button itself, which I think could be solved by making the width of the boxes slightly larger.

**Comment by DeadKper** [comment]
@dragos-efy 
> @DeadKper have you tried doing it in a private tab?

![image](https://user-images.githubusercontent.com/35360914/201534778-82c58e44-167b-4e1b-836a-1d0f7aa37809.png)

That's strange, firefox works now after deleting everything and restoring preferences, private tabs work also completly fine on both firefox and brave, but normal brave still doesn't load videos unless I ctrl + f5, but at least now loads with ctrl + f5, any ideas what the problem might be?, pd: the problem only happens when opening videos on a new tab (which I usually scroll trough my feed and open a new tab for every video I wanna watch)

**Comment by dragos-efy** [comment]
@photochlorination
> could be solved by making the width of the boxes slightly larger.

I know it's width, but you gotta consider ALL sorts of devices, like phones, tablets, tvs, desktops etc, including stuff like the galaxy fold, so making the width larger means less support for some devices, plus less cards shown in a row. Also consider vids with a time of `10:00:00` + views like `300k` + `watched`. I could add a breakpoint probably, so I'll experiment with that, but I'll see how it goes in testing, cuz I can't tell without trying first and rn I'm supposed to be on a mini break for 1 week.

**Comment by dragos-efy** [comment]
@DeadKper cool, yeah then we're talking about the same problem. I haven't had time to look into it and I'm supposed to be on a mini break for 1 week, so I guess my assumptions about why it happens are the same as in my last reply. I'd bet on the vue component being at fault. When I get time I'll try to replicate it locally so that I can fix it quickly, cuz it's indeed annoying. I also open vids in multiple tabs, since it's more efficient haha

**Comment by PrivacyDevel** [comment]
I tried to build the newest version of the efy branch just now because I discovered a bug and I wanted to update first before reporting a possibly already fixed bug.

Unfortunately that didn't work. `yarn run build` errors out with the following message:

```
[vite-plugin-pwa:build] '$ready_once' is not exported by efy/efy.js, imported by src/piped.js
file: /var/www/Piped-EFY2/src/piped.js:1:17
1: import {$, $all, $ready_once, $insert, $add, $append} from '../efy/efy.js';
                    ^
2: 
3: $ready_once('#efy_sidebar').then(()=>{
error during build:
Error: '$ready_once' is not exported by efy/efy.js, imported by src/piped.js
```

I manually checked out the `main` branch in the efy directory and did a `git pull`, hoping that this might solve it but to no avail. The same error message persists.

And the bug I discovered was that likes get displayed twice for comments. Like this: 
![comment](https://user-images.githubusercontent.com/105459436/202102746-925faa1a-fc4a-4f66-a7a9-17b2108c3a80.png)

**Comment by dragos-efy** [comment]
@PrivacyDevel hmm interesting. I fixed myself the double likes, which got introduced as a bug when the main got merged with efy, BEFORE my last update... and there seem to be some caching issues so I'm wondering if any of that is correlated... Looks like some issues disappeared after I changed my instance from "custom" to the official one... I think the move to piped.video messed up something for me personally... not sure if it's the same for others... Either way the bug about the variable is different most likely, but I'm just spitting my thoughts out haha. Feel free to dm me as well.

**Comment by dragos-efy** [comment]
@FireMasterK
> Webm/Opus could be used instead of mp3?

Done! They're all `webm` now. Only saved around 8% of space, but it's something lol

**Comment by dragos-efy** [comment]
@PrivacyDevel try updating the efy github submodule as well as piped. That might fix your problem

**Comment by PrivacyDevel** [comment]
@dragos-efy Yup, now it builds and the bug has been fixed. Thank you! :)

**Comment by dragos-efy** [comment]
@PrivacyDevel Well @Bnyro figured it out, I just had to remember what he did lol. Also dmed you on matrix, but you saw the reply here first

**Comment by PrivacyDevel** [comment]
@dragos-efy oh yeah sorry, I didn't notice that somehow.

**Comment by dragos-efy** [comment]
no worries

**Comment by cocoanaut** [comment]
Noticed the "Minimize Comments by default" toggle was not saving and have created a small PR to fix: https://github.com/TeamPiped/Piped/pull/1784.

**Comment by RaptaG** [comment]
Is merging the EFY PR planned soon?

**Comment by Bnyro** [comment]
> Is merging the EFY PR planned soon?
>
There's no ETA, it's going to be merged as soon as all bigger issues got resolved and it hence can be called stable and fast enough for replacing the current interface.

**Comment by lothar-cell** [comment]
This is amazing, can't wait for the final version!

**Comment by DeadKper** [comment]
Small suggestion, since vue.js sends a blank page with a div (or another container) with the "app" id to mount the page, shouldn't the white flash be fixed by just making the body have a dark background? just changing the style should be enough to something like:
<body style="background-color: #23272e;">
and then that can be replaced by de css of vue once the app loads since it will take all the screen, and a "dark" flash is not really a flash, don't think anyone will complain about it

**Comment by DogCatPuppyLover** [comment]
Could there be an option to remove the "you're offline" "you're back online" messages? They get can get annoying when you leave the page for a few hours and you have to close all of them.

**Comment by dragos-efy** [comment]
@cocoanaut Thank you!! 🔥 Saw it right away, but I was too lazy to reply lol

@DeadKper fixed in the next update. The solution is similar to what you described, but I made it work at the efy level so that all apps using it can benefit from no white flashes.

@DogCatPuppyLover yeah, I'll have section dedicated to notifications in the sidebar menu where you can choose the position (top left / right / center, bottom / left / right / center) and type of notifications you want shown / hidden, as well as a "missed notifications" number indicator that opens up the history, like on ios / android. It will probably take some time tho, since there's other stuff to take care of first, but yeah, I agree, that should be controlled by a toggle. I'll also make them expire after a few seconds (can be chosen by the user), which moves them to the notification history.

**Comment by Konglomneshued** [comment]
I don't know how, but I used an instance not listed in the Github and it uses the new EFY UI.

https://piped.mha.fi/

Problem is, that instance is really buggy and it links to this Github, so I can't submit an issue.

**Comment by Konglomneshued** [comment]
https://github.com/TeamPiped/Piped/issues/1391#issuecomment-1297487894
> People! is it just me or do you also hate the way the player situation is in ios , seriously who wants this
> 
> EDIT: I was to ask are there any plans for changing it ? or is it just way it was intended to be , because it is hard for me to navigate the player UI and I don’t see the point of having two players , rather the native iOS Safari player can be used for things like subtitles and playback speed , as for the quality change button it can have a dedicated button , this is the best I can say as I’m not a dev
> 
> Thanks!
> 
> ![2838F4CC-F08A-4744-B1BC-C69F0A018A2B](https://user-images.githubusercontent.com/79302042/199080856-59c3748d-f8b1-4252-a257-3a469f75569f.png)


I think this is related to `min-height: 100vh`

See: https://ilxanlar.medium.com/you-shouldnt-rely-on-css-100vh-and-here-s-why-1b4721e74487

**Comment by dragos-efy** [comment]
@Konglomneshued
> instance is really buggy

Say what's buggy specifically and take screenshots, copy errors etc, otherwise I don't know what your bugs are. Nice you included a link and a possible cause tho.

> Looks like this was already talked about here #1931

So I have no clue what your issues are, but if it's related to that comment, that was NOT the new efy redesign, that screenshot has the old design in it, therefore wrong place to ask here about that. If it's something else that is actually related to efy, then I can help, but send me screenshots or give me more details, otherwise I'll have to ignore it and focus on stuff that's actually part of the redesign. Thanks!

Also keep in mind that the mha.fi instance is not in our control, the owner took BETA versions that are NOT ready to be used by all users yet from the efy repo automatically I'm guessing. While most users of that instance are happy it seems, it was too early to release it, we actually discouraged instance owners from switching to it, but they did cuz they liked it.

If you wanna test the latest changes, then use [efy.piped.pages.dev](https://efy.piped.pages.dev) , which will have an update later today, but you'll have to clear your cache cuz it's a huge update with breaking changes. I'll post the instructions right after the update today here as well as in the [piped matrix room](https://matrix.to/#/#piped:matrix.org) and [efy matrix room](https://matrix.to/#/#efy_ui:matrix.org)

**Comment by dragos-efy** [comment]
IMPORTANT! You'll likely have to do one or both of these, because the new update can potentially break some stuff:

1. `efy sidebar menu` > `backup` > `efy settings` > `reset`
2.  `ctrl + f5` or clear your cache / do a hard refresh

You'll know that you're using the new update if you go to `efy sidebar` > click the `efy logo` > see the version `2022.12.15 Beta`

**Comment by Konglomneshued** [comment]
> IMPORTANT! You'll likely have to do one or both of these, because the new update can potentially break some stuff:
> 
>     1. `efy sidebar menu` > `backup` > `efy settings` > `reset`
> 
>     2. `ctrl + f5` or clear your cache / do a hard refresh
> 
> 
> You'll know that you're using the new update if you go to `efy sidebar` > click the `efy logo` > see the version `2022.12.15 Beta`

I was about to close the issue I created, but I noticed the same problem I had using using https://piped.mha.fi/ when I tried the [efy.piped.pages.dev](https://efy.piped.pages.dev/) link you provided.  I thought it might help to keep it there, so it can be fixed.

**Comment by IkelAtomig** [comment]
> @photochlorination I could add a minimum height which makes them all the same, but then all cards would be tall even when not needed. I'm keeping this on hold for now unless you have practical ideas to suggest, although if there's enough interest, I can maybe add an alpha testing option in the sidebar for users to try out multiple variations. Thanks!

Dragos, Better put the duration over the thumbnail's bottom right and Shorts tag in Bottom left. That would do better.

**Comment by IkelAtomig** [comment]
Full width comments are not being saved.

**Comment by IkelAtomig** [comment]
Would be better if nature effects is around the video player and not touching it.

**Comment by IkelAtomig** [comment]
![image](https://user-images.githubusercontent.com/101938856/209513029-412d7f34-f9a9-4a27-9441-49d3447bc777.png)

This is for https://efy.piped.pages.dev/playlists

As it is already under playlist section. Remove the word `Playlist`. So, They are in single line. Reduce space again.

**Comment by IkelAtomig** [comment]
![image](https://user-images.githubusercontent.com/101938856/209513406-47219c9e-9e7b-462a-afed-eb95289bcc6c.png)

In trending there is additional paramater. Date. Which could be moved up to 1st line when If time is placed over thumbnail.

**Comment by IkelAtomig** [comment]
I think instead of separate borders for all the elements under the video Name. Make it a transparent border. Better make it to be consistent across all video cards. 

Maybe give an option to decrease the card size. So, there can be many videos in a single row. But this would be efficient only if any of the above is done.

**Comment by PrivacyDevel** [comment]
Nothing happens anymore when you left-click a video thumbnail in the new EFY UI 2022.12.15 Beta. You have to click on the video title if you want the watch page for that video to open. That didn't use to be the case in the previous version.

**Comment by dragos-efy** [comment]
@PrivacyDevel @kittyra didn't know how to replicate it, but I found out it worked fine from me cuz I was on touch screen inputs, so only the cursor seems to be affected by it. Either way, it looks like the `<a>` element that includes the thumbnail and title of the video doesn't match the height, so then the thumbnail works only in the bottom area of the thumbnail. I'll add `display: block` to it in the next update and that should fix it, it works well now with the modification. Thanks for the heads up!

**Comment by dragos-efy** [comment]
@IkelAtomig 

> Remove the word Playlist

Thought of that too, been doing it in multiple places already. It's also good for reducing the amount of translations needed.

> Maybe give an option to decrease the card size

We have custom width in alpha, which changes the amount of cards. Not the same, but similar for now.

> Full width comments are not being saved

Known issue, that's why it's in alpha. Alpha should be treated as alpha, as in broken, barely functional. It's more for letting curious users test features ahead and give feedback before becoming stable and safe to use for most or all users.

> nature effects is around the video player

I remember this, but I'm rebuilding the logic for them to use webp, avif, png, jpg etc and become more efficient. There will be multiple pseudo 3d layers behind and in front of the app, so I'll first make that happen to not waste time on implementing something, then needing to rebuild it anyway.

And for video cards, I'll add an option to choose what info to display + extra styling specifically for that area that's customised by the user. I don't think there's a best design there and each person seems to have a different take, so to make it useful and a great experience for all, I'll make it more modular and let people decide how they want it to look like. Not sure how hard it will be to implement it, but at least half of it should be easy. Positioning can be a bit more challenging, but we'll see.

**Comment by dragos-efy** [comment]
@PrivacyDevel @kittyra thumbnails on firefox should be fixed on version 23.01.07 and above!

@IkelAtomig I renamed the text for playlist options, however it's only for English rn cuz it's not the official branch. Your name is updated btw, did that some time ago.

**Comment by dragos-efy** [comment]
Quick heads up! On version 23.01.13 beta and above, reset your background images if you saved any: `efy sidebar` > `backup` > `efy images` > `reset`. The new update has a way more efficient way of handling thumbnails than previously, but your current ones will likely break until you do a reset.

**Comment by dragos-efy** [comment]
@eladkarako Thanks for the feedback! Glad you like it! For lazy loading, it's technically not part of my responsibilities (unless it became a thing in main recently cuz we're out of date by a bit), but if it ends up working well I can do it, I'll try it out. And about haptic feedback, I wasn't aware it can be controlled from the browser, that's cool, more immersive and it seems simple to implement... safari and firefox mobile seem to not support it, but I guess I can make it a toggle in the efy menu and only people who want or can use it would do. And if you wanna control the animation btw, you can change the speed from `efy` > `accessibility` > `animations`, although some effects will probably work best on default, and short clicks will be too short to see the difference, since it's pseudo pressure based, the longer you tap / click the longer it holds it, but still, it's there if you need it. And the focus outline will be on only when turning on the accessibility feature in the next updates, to make it more consistent.

**Comment by Imzxhir** [comment]
I keep getting an issue on Firefox where the checkboxes don't render correctly, I am not sure what the issue is and I even tried on a fresh profile on Firefox. On a different browser like Chrome, the checkboxes all work and render correctly. This could be an issue with browsers using a different engine, Firefox using Gecko and Chrome using Blink as their browser engine.

Firefox:

![image](https://github.com/TeamPiped/Piped/assets/98621617/fe0bcc5d-5cfc-4773-b4e2-ee06cda4957f)

Chrome:

![image](https://github.com/TeamPiped/Piped/assets/98621617/da6e6f12-0dcf-439c-844d-9b297e0d7c70)

**Comment by dragos-efy** [comment]
@Imzxhir can you try it on firefox but on a different device and in a private window? like another pc, laptop etc? I'm trying to replicate you problem but it works fine for me, so my first thought is that maybe you have an extension that interferes with it, or some firefox profile / setting that is responsible for theming stuff (since firefox has some options for forcing a certain css styling, which can become kinda annoying when it doesn't work as it should and leads to broken pages, weird colors etc).

Unfortunately, since I can't replicate it, you're gonna have to give me more context or try finding a pattern that might be causing it, but thanks for the heads up and lmk how it goes

**Comment by Imzxhir** [comment]
It seems to render the checkboxes correctly on firefox for mobile which makes me to believe that it is not a render engine issue.

I have created a new profile on firefox which removes all extensions and settings from firefox and sets the defualt vaules as if you were installing firefox brand new, and I have still got the issue. I'll try to see if I can find ways to replicate the issue / bug.

**Comment by dragos-efy** [comment]
cool. I'll probably switch to a font icon based approach for the checkbox anyway eventually, since the current way is sort of a mini css hack rather than a proper way, but getting this figured out would be nice

**Comment by DeadKper** [comment]
LBRY videos don't load on efy, works normally on piped.kavin.rocks, I'm using brave, tried without extensions and it's the same result, with LBRY streaming disabled the error just changes
LBRY streaming enabled
![image](https://github.com/TeamPiped/Piped/assets/35360914/5ffbdf5b-d37c-4f83-b200-ad1d36c7d430)
LBRY streaming disabled
![image](https://github.com/TeamPiped/Piped/assets/35360914/f40e2fe0-e4ac-40df-8ddc-be6e8d2e4e06)
also tried deleting cookies but it's still the same, I was testing on efy.piped.pages.dev

**Comment by dragos-efy** [comment]
@DeadKper Yeah, same for everyone, probably cuz the efy version isn't using some newer code from piped. I'll update it in the next 1-2 weeks, maybe that will fix it. Thanks!

**Comment by IkelAtomig** [comment]
> Thought of that too, been doing it in multiple places already. It's also good for reducing the amount of translations needed.

![image](https://github.com/TeamPiped/Piped/assets/101938856/9bc9f853-8dab-47ed-bf02-fb7cbca699b3)

I would suggest replacing the delete button with a trash Icon and on mouse hover reads as 'Deletes Playlist'

**Comment by Konglomneshued** [comment]
Having the mouse over defeats the purpose of replacing the text with an icon to reduce the translations needed.

**Comment by dragos-efy** [comment]
also, people using touchscreen can kinda hover if they hold the pen or finger, but it's not really a convergent solution, cuz it's messy to do it on touchscreen. I wouldn't say it completely defeats the point tho if the goal is reducing space, that could be good, but for translations, yeah, no point.

There's a good reason for titles on hovering tho, accessibility and text readers. When I use buttons with icons only, I'm trying now to kinda always add titles.

We could also make it a split option, whoever wants or doesn't want text they can toggle it. But idk yet, I'll have to think more about it... we could also ask more users as well or do a poll, but `rename` and `delete` could be just icons in theory with titles for accessibility or hovering if the user has a mouse / pen and then maybe in the menu add a toggle for text if users want that and there's a need.

**Comment by IkelAtomig** [comment]
Dragos, that seems a pretty good idea.

**Comment by julianfairfax** [comment]
It would be nice if there was a "follow the system theme" option for the theme

![image](https://github.com/TeamPiped/Piped/assets/43753131/f6ffd249-9c6e-42ef-9fc1-017680d9a7c6)

**Comment by dragos-efy** [comment]
@julianfairfax makes sense. There's a problem tho, light & dark themes are usually not very customizable like with efy, which is why that concept works, cuz they don't have to consider transparency, potential background images etc, but in order to do something similar in efy I'd have to let the user store 2 different themes at the same time and based on the time of the day or browser's preference change the one or the other.

I could just toggle the light or dark mode, but if the user modifies the colors of the text and background for example, it can be kinda weird for the user to be switched back to the default or to see "broken colors", so I pretty much have to go with the approach mentioned above, because efy has billions of themes, not just 2 like normal frameworks.

Also, in efy you can import and export themes from `efy sidebar` > `Backup` > `EFY Settings`. What's cool about that is that you can also have different sounds settings, effects, colors, gaps, radius etc based on the file. So if you create a light and dark backup for each, you can technically just import the one you need quickly in 4-5 clicks, but you also get the benefit of configuring more stuff than just the colors.

I'll eventually add a feature to automatically load a saved configuration based on a trigger like time of the day or the browser's theme preference, but for now you gotta do it manually, cuz that will be complicated to implement and it will probably take a long time, considering that I have lots of projects and features to work on.

**Comment by dragos-efy** [comment]
@eladkarako thanks for the detailed feedback. Yeah, I'm working right now on updating it to the latest Piped version, since  it currently uses an older version. As for the cards and labels, I think it makes sense to put more emphasis on the checkboxes and labels than the container, as you said, so I'll look into that. The hover effect you did is interesting, but on touchscreen that won't work, unless using a pen, so the solution needs to cover fingers on touchscreen too, or a design that will let you know stuff without having to hover on it.

And for the part about lazy loading stuff it kinda makes sense to me, but I'm more in charge of design than that sort of stuff so maybe @FireMasterK or someone else would be better at understanding and deciding on that, since I think it applies to Piped in general, rather than efy or design specificially.

Anyway, I'm gonna first release the latest Piped + latest efy to make sure that stuff is up to date and then I'll work on issues and features. Yours is medium priority I think, there's higher priority ones to deal with first, but I'll get to it as soon as my time allows it. Thanks again and if you have more input, feel free to share it!

**Comment by dragos-efy** [comment]
New update is live at [efy.piped.pages.dev](https://efy.piped.pages.dev) 🥳 I fixed some of your issues too, but it's more of a maintanance and compatibility update + loads of efficiency improvements and efy features, so expect the newer updates to have more features and bug fixes. Keep the feedback coming, it's been very useful. We also talk, do polls etc on the matrix room from time to time. Enjoy!

**Comment by Donty10** [comment]
Not sure if this is the right place to ask, but how are you meant to install this? I had done it previously by building the docker image from the repo, but now I am getting errors. Is there a pre-built image anywhere that I can use?

**Comment by unstablemaple** [comment]
I as well can't build a docker image for piped-efy. I get this error:

```10.41 Error: No ESLint configuration found in /app/src.
10.41     at CascadingConfigArrayFactory._finalizeConfigArray (/app/node_modules/.pnpm/@eslint+eslintrc@2.1.2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3989:19)
10.41     at CascadingConfigArrayFactory.getConfigArrayForFile (/app/node_modules/.pnpm/@eslint+eslintrc@2.1.2/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3780:21)
10.41     at CLIEngine.isPathIgnored (/app/node_modules/.pnpm/eslint@8.48.0/node_modules/eslint/lib/cli-engine/cli-engine.js:1000:18)
10.41     at ESLint.isPathIgnored (/app/node_modules/.pnpm/eslint@8.48.0/node_modules/eslint/lib/eslint/eslint.js:681:26)
10.41     at Object.transform (/app/node_modules/.pnpm/vite-plugin-eslint@1.8.1_eslint@8.48.0_vite@4.4.9/node_modules/vite-plugin-eslint/dist/index.js:1:2348)
10.41     at file:///app/node_modules/.pnpm/rollup@3.28.1/node_modules/rollup/dist/es/shared/node-entry.js:25461:40
10.43  ELIFECYCLE  Command failed with exit code 1.```

**Comment by dragos-efy** [comment]
@Donty10 @unstablemaple Thanks guys for letting us know, I don't know docker, so maybe @FireMasterK can give more insight? (if you have time)

**Comment by Donty10** [comment]
Any update on this? Did you find a workaround?

**Comment by IkelAtomig** [comment]
Really good work, dragos. The feed is now good and display 5 videos in a row as in original. Much appreciated for that as it is much needed. For piped style, It would be lot better if it enters beta stage.

**Comment by dragos-efy** [comment]
Thanks! I'll add some new stuff specific to Piped soon. Also noticed a few small bugs I gotta fix, but yeah, overall it's getting pretty stable except for the piped section in the menu (which will have some of the new stuff)

**Comment by BarbzYHOOL** [comment]
can we edit the theme as simple user or only if you host your own piped?

Also, I wanted to create a new issue but the default dark theme is hard on the eyes. The github dark theme is better for example.
You can use this application on linux "Kontrast" that helps to find good contrast. Actually it's too dark with white text

**Comment by dragos-efy** [comment]
@BarbzYHOOL Yeah, [efy](https://efy.ooo) (the UI framework it uses) is very customizable for the user, you can create billions of designs, just click / tap on the sidebar menu button from the top right corner (3 horizontal lines like this: `≡`) and under `theme` you should see options for colors, including buttons, text, border, background, shadows etc. Kinda funny you asked that, since that's what it focuses on, but maybe you haven't tried the menu yet haha.

In fact, if you like the github theme, you can probably paste the colors they have into efy, change the radius, borders etc and it will look similar. And not only that, but if someome else wants the same, you can share the file with them.

You can also export a theme as json from `efy menu` > `backup` > `EFY Settings` > `save` and edit the json in a text editor, then when you're done you import it, so you can do it visually or in a text editor, both ways work, as long as you use the right values (which I will create documentation for in the future).

Just keep in mind to backup your theme, cuz this Piped instance is not fully stable yet and soon I will push a breaking update.

**Comment by BarbzYHOOL** [comment]
I am using public Piped instances and there is no such option example here https://piped.lunar.icu/
can only change to dark or white theme

Is it because they haven't updated?

**Comment by dragos-efy** [comment]
@BarbzYHOOL those instances don't use efy, that's why, there's 3-4 afaik, but the most up to date one is [efy.piped.pages.dev](https://efy.piped.pages.dev), so I'd recommend that one. As for the source code, you can see it on the efy branch of Piped. It's not part of the main branch yet to smooth out issues, stability etc before making it official. It's still in the experimental phase if that makes sense.

**Comment by BarbzYHOOL** [comment]
> @BarbzYHOOL those instances don't use efy, that's why, there's 3-4 afaik, but the most up to date one is [efy.piped.pages.dev](https://efy.piped.pages.dev), so I'd recommend that one. As for the source code, you can see it on the efy branch of Piped. It's not part of the main branch yet to smooth out issues, stability etc before making it official. It's still in the experimental phase if that makes sense.

thanks, should i only use this piped instance for testing?

PS: the piped instance i'm using has trouble loading videos, kinda annoying, i hope it's just the instance, i'm new to it

**Comment by dragos-efy** [comment]
@BarbzYHOOL you can use other ones too, but you're not guaranteed to have the latest version, so for that, yeah, that instance is the most recommended and "official" haha.

For loading videos, Youtube bans IPs or changes its structure constantly, which we're not fully in control, change the backend instance from settings when you notice that until you find one that works. You can keep the same frontend, you just gotta change the backend basically. But that's related to Piped in general, not this instance.

Anyway, let me know how it goes with theming if you have thoughts you wanna share, since this thread is for that. You can also ask efy questions in the [efy room](https://matrix.to/#/#efy_ui:matrix.org) or Piped questions in the [Piped room](https://matrix.to/#/#piped:matrix.org) depending on what the questions are related to.

**Comment by BarbzYHOOL** [comment]
one of the theme setting ends up like this, and then you cannot fnd the menu button back

![image](https://github.com/TeamPiped/Piped/assets/29780637/1b87b9a5-d7c2-4c9b-bfc9-7c5394b930e1)

**Comment by dragos-efy** [comment]
@BarbzYHOOL it's cuz either your browser and desktop environment doesn't support window transparency or you need a transparent gtk / qt theme. It's explained on the question mark alongside. You can use transparency on any device tho if you add your own images from the `add file` button, which in the next version will be changed with something like this:

![image](https://github.com/TeamPiped/Piped/assets/86549690/748d94c8-e57e-461f-b4f0-2f4a5330af66)

And in case it's not clear, transparent windows look like this (ignore the red line, I'm too lazy to take a new screenshot lol):

![image](https://github.com/TeamPiped/Piped/assets/86549690/75f879d9-0087-44b6-b114-e1ab5ceeaf2f)

while background images with transparency or layers like this:

![image](https://github.com/TeamPiped/Piped/assets/86549690/f83b26f4-d160-41f3-98dc-ff1234ea5344)

The window addresses what's behind it, while the background what's above it.

**Comment by kzhe2059** [comment]
Major bug: all images disappear when I log in.

Without logging in on smnz.de, I see this.
![Screenshot from 2023-12-04 18-45-31](https://github.com/TeamPiped/Piped/assets/29979557/b86d5f77-3ac4-4851-a9f8-7b3f4fa69894)
After log in, I see this.
![Screenshot from 2023-12-04 18-46-43](https://github.com/TeamPiped/Piped/assets/29979557/45daac81-c31c-4cd7-9d0e-7e2a335288c2)

**Comment by dragos-efy** [comment]
@kzhe2059 Thanks for the feedback! For me it works fine on other instances than smnz.de, so it doesn't seem to be an issue related to the efy instance, because on piped.video for example it does the same, so I'd suggest you to switch the instance or contact the instance owner if they're okay with that

**Comment by kzhe2059** [comment]
Is there any way to use a custom instance on the new UI?

**Comment by dragos-efy** [comment]
@kzhe2059  If you mean changing the backend instance, then go to `settings` > `instance`. The custom authentication instance is below as well.

![custom_instance](https://github.com/TeamPiped/Piped/assets/86549690/1aca26db-0d1a-415f-85e8-814099243669)

**Comment by kzhe2059** [comment]
Yes, but you can only choose from the list, not input custom url for the api and such

**Comment by Reinachan** [comment]
I figure this is where feedback on the redesign goes?

Why is everything in video cards styled like buttons? This feels really excessive and makes it really unclear what is or isn't a button within the UI.
![screenshot of a video card](https://github.com/TeamPiped/Piped/assets/16106839/d5054b78-b25d-45e5-abdf-b3c09d1b2bbe)

I'd suggest going back to the style used in old Piped for this. Like, you can put it all into a card (though I personally prefer it it isn't), but the way the items within were laid out in old Piped worked. Changing this feels like change for the sake of change. 
![screenshot of the same video but in old piped](https://github.com/TeamPiped/Piped/assets/16106839/d84ae6b5-a094-4ebf-a9a1-ae98f7e206d0)

The old video preview layout actually worked so well that when YouTube changed it, I coincidentally userstyled it to look almost the same as Piped before I even knew of Piped xD

I can understand making some changes for the sake of improving usability on touch devices. But I'd suggest at least de-buttonify the stats and the channel name and move the stats back under the channel name again. Also please move the timestamp back up to the thumbnail.

Other than that, having the search bar and every kind of card and input change its scale when you click on them is fairly distracting. It can make sense for buttons but I wouldn't use it for everything. It makes interacting with the customisation options in the sidebar really whack too.

**Comment by dragos-efy** [comment]
Hey @Reinachan! Yeah, it's the right place.

1. Card layout - there will be multiple layouts... so we'll technically have the new and old one as well, chosen by the user from the menu. Also the reason I went with this new approach is to allow users to choose what to show or hide (time, views, add to playlist, audio version, time ago, etc) and to also have a more "breathable" design that makes it easier to spot information. But yeah, no worries, you'll be able to switch to the other layout too and whatever layout you like, both should be customizable. It's more of a problem of time, since I was very busy with lots of projects, work, finding jobs, life etc, and while I prefer the new layout personally, I agree with your points as well, especially cuz other people from the community brought this up as well, so expect this stuff to be added when I have time.

2. Click animation - I agree, in the future you'll be able to turn it on / off, also individually by type (buttons, inputs, etc), in a similar way where now for example you can change the speed. Which speaking of... when it's 0, it completely turns off all of them in newer efy versions (already implemented), but Piped didn't get that update yet.

3. "Userstyling" YT - that's pretty cool, before working directly on Piped I also injected css and js on my own browser, and before that for youtube as well lol. So I understand your mindset perfectly, which is why I made the efy integration with Piped so customizable... to allow nerds like us to do wild stuff, with or without coding, directly from the app, and that's the main priority, giving more power to the user.

Thanks for the feedback, feel free to give me more whenever you feel the need to, it helps me and the community overall have a better experience! Just keep in mind that my time for Piped is limited currently cuz of other priorities, but soon I'll work on more stuff

**Comment by dragos-efy** [comment]
@kzhe2059 that's not related to efy then... cuz the official piped doesn't have that feature either, so I guess maybe talk to the main devs about it IF THEY LIKE THE IDEA, else it won't be added in the efy version either, since I'd have to rely on the defaults, which I'm not responsible for.

**Comment by Imzxhir** [comment]
Hello! I noticed that there is no share video button. Is that planned to be added?

**Comment by dragos-efy** [comment]
@Imzxhir yes, it's a bug I'll fix soon, it should be easy, it annoys me too, as someone who copies links a lot haha

**Comment by athiosh** [comment]
how can i install this locally? Is there documentation for install or a docker compose?

**Comment by dragos-efy** [comment]
@athiosh You can run it locally if you clone it and switch to the efy branch, or we have a docker tutorial on the efy site:

## 1. Local
Clone it from github:
```
git clone https://github.com/TeamPiped/Piped.git
cd Piped
git switch efy
```
To run it first install packages with:
```
pnpm install
```
And then you can use either
```
pnpm dev
```
or
```
pnpm build
pnpm preview
```
In the terminal you'll see the link you gotta navigate to in your browser

---

## 2. Docker
Follow the tutorial at [https://efy.ooo/#faq#piped_instance](https://efy.ooo/#faq#piped_instance)

---

Now that you mentioned it, maybe I should add the 1st method to the docs as well when I get some spare time 🤔

**Comment by dragos-efy** [comment]
@Imzxhir forgot to reply here, but the bug seems to be caused by uBlock Origin, so if you have it on, try turning it off only for piped or changing your configuration maybe, that worked for me and other people I've been talking to. I'm still looking for other methods tho, but to keep you updated on it

**Comment by athiosh** [comment]
@dragos-efy  thank you for the documentation. Awesome work.

**Comment by athiosh** [comment]
After some testing, I redeploy this to my production server and I found that none of the pages worked If I start fresh without cookies or cache, I had to use the original frontend image to login first in my browser then change to efy frontend image.
 
<img width="1489" alt="Screenshot 2024-07-12 at 10 08 03 PM" src="https://github.com/user-attachments/assets/d6c65878-aa94-4105-89a1-ff8d6e7ce39d">

**Comment by dragos-efy** [comment]
@athiosh so wait, to confirm, does it work now after doing that? And no worries, glad to help!

**Comment by dragos-efy** [comment]
@codenyte cool! I added it to the [project's tasks](https://github.com/users/dragos-efy/projects/2/views/5)

**Comment by athiosh** [comment]
@dragos-efy , it only works if the browser is cached with my login session. If I wasn't login prior to running the efy image it will be blank like the screenshot I shared. I hope this make sense.

**Comment by dragos-efy** [comment]
@athiosh thanks 🤟 it makes sense, yeah, I wanted to be sure I fully get you & I added it to the [tasks](https://github.com/users/dragos-efy/projects/2/views/5), but keep in mind that I'm an absolute noob at docker and I have other priorities to work on, so unless someone else finds a fix, it will probably take me a while to dig deeper into it. I'm glad you mentioned it tho, hopefully we'll find a solution

**Comment by athiosh** [comment]
@dragos-efy I've spent sometime today trying to figure out what is causing the issue and after reading the error carefully it shows that page failed to construct the URL, so after some more investigation I was able to make the page load after adding instance and auth_instance_url variable in the local storage in the browser. Once those two value is populated, everything worked. The first screenshot is the current state where those variable is not populated on start. The second screenshot is where I have manually enter the variable, ignore the authtoken as it generate after login. I hope this would help in fixing the issue.

<img width="1470" alt="Screenshot 2024-07-14 at 2 57 15 PM" src="https://github.com/user-attachments/assets/ef7eccc4-242b-4140-9784-0f4c3154eb07">

<img width="1449" alt="Screenshot 2024-07-14 at 3 02 04 PM" src="https://github.com/user-attachments/assets/8a7b8a25-6390-4da9-9790-f26f3b35ae04">

**Comment by dragos-efy** [comment]
I think the auth variables are a pretty significant clue for sure! Thanks for the info & screenshots 👏

**Comment by dragos-efy** [comment]
@athiosh I think that the auth variables are a pretty significant clue for sure! Thanks for the info & screenshots 👏

**Comment by athiosh** [comment]
Recently, I tried to enable caption on videos but every time I enable it, it will crash as shown in the screenshot. I tried reverting to the official frontend and no issues there.

<img width="680" alt="Screenshot 2024-08-27 at 3 00 59 PM" src="https://github.com/user-attachments/assets/8f910537-6414-4dd7-b454-328165169375">

**Comment by dragos-efy** [comment]
@athiosh well I can't test it rn cuz all public instances are blocked by yt. Are you doing that on a private instance? Or are some instances working that I'm not aware of?

**Comment by athiosh** [comment]
yea, i am hosting it privately using docker.

**Comment by vayan** [comment]
> I was able to make the page load after adding instance and auth_instance_url variable in the local storage in the browser

That or you navigate to https://piped_instance.example.com/preferences and set the custom instance there.

Is there a way to set or build EFY with a default instance?

**Comment by athiosh** [comment]
> @athiosh I think that the auth variables are a pretty significant clue for sure! Thanks for the info & screenshots 👏

@dragos-efy , I had some time yesterday and decided to take a stab at this. The issues seems to be the Dockerfile not getting the .env file during build, I just hardcoded it inside the Dockerfile after the WORKDIR /app/ line and it worked after.

`
ENV VITE_PIPED_API=https://pipedapi.<redacted>.com
ENV VITE_PIPED_PROXY=https://pipedproxy.<redacted>.com
ENV VITE_PIPED_INSTANCES=https://piped.<redacted>.com
`

**Comment by dragos-efy** [comment]
> Is there a way to set or build EFY with a default instance?

@vayan the efy branch is the same as main but with some changes to the vue components + adding the efy package so that those components apply the changes related to efy visually, so the answer is YES, you just modify parts to be compatible, which is what we've been doing all this time. I haven't gotten a lot of time for it lately tho and the problem is that since the efy branch is not the official one, there's always a TON of files that need to be modified to keep up and have stuff updated, so it's quite time consuming and annoying to do all of that regularly, which is why I proposed to either make efy the default or just have efy as a different thing that is independent from the main branch, cuz it's just frustrating to manage it all as mainly a 1 person activity who has over 20 other projects to work on as well apart from this.

Short answer, you can modify the config files or vue files directly and the build that and it should work, it's just a bit time consuming to solve all the mini errors that show up when you try to do that, but the process itself is kinda easy

**Comment by dragos-efy** [comment]
@athiosh that's coooooool 🔥 Thanks for sharing, I bet that some lurkers who don't comment publicly but look for solutions will find this useful too haha 👀

**Comment by athiosh** [comment]
@dragos-efy , one thing left to fix is the caption. The page will crash everytime i enable caption in chromium based browser on my macbook however every other browser like Safari and firefox will work fine even chromium base ios browser has no issue either. Do you have any thoughts on what might cause the issue?

**Comment by dragos-efy** [comment]
@athiosh no clue, maybe try turning off extensions, or try chromium on linux / windows to see if it's the same, then if it's the same on those it's probably chromium related, if not, then mac related. Chromium on ios uses webkit anyway, not v8, so it's not really the same chromium as on linux, windows, mac, etc, therefore it's kinda like testing on safari, so it doesn't count
